### PR TITLE
remove dependency on usocket

### PR DESCRIPTION
**Pull request by jmercouris**
it is enough to test a hostname with iolib. that said, I do not believe we will be able to easily get rid of iolib...!

### CODE REVIEW

**Comment by Ambrevar** [comment]
Good catch! I didn't know iolib could do this!

**Comment by Ambrevar** [comment]
Tests are failing though, so this might not be completely equivalent.  Can you investigate?

**Comment by jmercouris** [comment]
Tests are now passing. It seems that usocket was also "allowing" IP addresses. I've since added a OR clause to check if it is an IP address.

File: source/urls.lisp [hunk 1]
```diff
@@ -42,23 +42,32 @@ On errors, return URL."
...
-              ;; get-host-by-name may signal a ns-try-again-condition which is
-              ;; not an error, so we can't use `ignore-errors' here.
-              (handler-case (usocket:get-host-by-name (quri:uri-host uri))
-                (t () nil)))))))
+    (flet ((hostname-found-p (name)
+             (handler-case (iolib/sockets:lookup-hostname name)
+               (t () nil)))
+           (valid-ip-address-p (address)
+             (cl-ppcre:scan
+              (cl-ppcre:create-scanner "\\d{1,3}\.\\d{1,3}\.\\d{1,3}\.\\d{1,3}")
```
Comment (Ambrevar) [inline source/urls.lisp:50]: Note: You could have removed `create-scanner`, just keep the string.
This is wrong though: it accepts characters before and after the IP address, so you need to put the regexp between `^` and `$`.

Actually, scratch all that: there is `quri:ip-addr-p` :)

File: source/urls.lisp [hunk 2]
```diff
@@ -42,23 +42,32 @@ On errors, return URL."
              "thismessage" "tip" "tn3270" "turn" "turns" "tv" "urn" "vemmi" "vnc" "ws" "wss"
              "xcon" "xcon-userid" "xmlrpc.beep" "xmlrpc.beeps" "xmpp" "z39.50r" "z39.50s")))
```
Comment (Ambrevar) [inline source/urls.lisp:43]: While we are at it, could you let-bind the Nyxt internal scheme to an explicit name (say "nyxt-internal-schemes"), same with the IANA schemes, then remove the comment "For Nyxt's internal URLs and not-standardised ones".

File: source/urls.lisp (hunk 1 above)
Comment (jmercouris) [reply source/urls.lisp]: Thank you!

**Comment by jmercouris** [comment]
Thank you for your feedback!

**Pull request merged**